tests/utils/gossamer_utils.go

```diff
@@ -314,61 +312,83 @@ func StartNodes(t *testing.T, nodes []*Node) error {
 }
 
 // InitializeAndStartNodes will spin up `num` gossamer nodes
-func InitializeAndStartNodes(t *testing.T, num int, genesis, config string) ([]*Node, error) {
-	var nodes []*Node
-
+func InitializeAndStartNodes(t *testing.T, num int, genesis, config string) (
+	nodes []*Node, err error) {
 	var wg sync.WaitGroup
-	var nodeMu sync.Mutex
+	var nodesMutex, errMutex sync.Mutex
 	wg.Add(num)
 
 	for i := 0; i < num; i++ {
 		go func(i int) {
+			defer wg.Done()
 			name := strconv.Itoa(i)
 			if i < len(KeyList) {
 				name = KeyList[i]
 			}
-			node, err := RunGossamer(t, i, TestDir(t, name), genesis, config, false, false)
-			if err != nil {
-				Logger.Errorf("failed to run Gossamer for node index %d", i)
+			node, runErr := RunGossamer(t, i, TestDir(t, name), genesis, config, false, false)
+			if runErr != nil {
+				errMutex.Lock()
+				if err == nil {
+					err = fmt.Errorf("failed to run Gossamer for node index %d: %w", i, runErr)
+				}
+				errMutex.Unlock()
+				return
 			}
 
-			nodeMu.Lock()
+			nodesMutex.Lock()
 			nodes = append(nodes, node)
-			nodeMu.Unlock()
-			wg.Done()
+			nodesMutex.Unlock()
 		}(i)
 	}
 
 	wg.Wait()
 
+	if err != nil {
+		_ = StopNodes(t, nodes)
+		return nil, err
+	}
+
 	return nodes, nil
 }
 
 // InitializeAndStartNodesWebsocket will spin up `num` gossamer nodes running with Websocket rpc enabled
-func InitializeAndStartNodesWebsocket(t *testing.T, num int, genesis, config string) ([]*Node, error) {
-	var nodes []*Node
-
+func InitializeAndStartNodesWebsocket(t *testing.T, num int, genesis, config string) (
+	nodes []*Node, err error) {
+	var nodesMutex, errMutex sync.Mutex
```

Comment:
Good point, although there is little point to do this considering the future-to-be changes: this parallel logic is changed to be sequential since it takes a short time to init+start, so only the first error encountered is returned.